### PR TITLE
Create logger wrapper

### PR DESCRIPTION
**Pull request by tomaszdurka**

### CODE REVIEW

**Comment by tomaszdurka** [comment]
@vogdb please take over

**Comment by vogdb** [comment]
@tomaszdurka please review

File: lib/cm-api-client.js [hunk 1]
```diff
@@ -36,7 +37,7 @@ CMApiClient.prototype._request = function(action, data) {
     json: true
   };
 
-  serviceLocator.get('logger').info('cm-api', 'request', options.uri, options.body);
+  serviceLocator.get('logger').info('cm-api request', {options: _.pick(options, 'uri', 'body')});
```
Comment (tomaszdurka) [inline lib/cm-api-client.js]: Let's pass all options?

File: lib/global/log4js-layout-json.js [hunk 2]
```diff
@@ -11,10 +11,12 @@ var Log4jsJsonLayout = {
       time: logEvent.startTime.toISOString(),
       level: logEvent.level && logEvent.level.levelStr || null
```
Comment (tomaszdurka) [inline lib/global/log4js-layout-json.js]:
```js
logEvent.level && logEvent.level.levelStr.toLowerCase() || null
```

File: lib/logger.js [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+}
+
+Logger.prototype._log = function(level, message, extra) {
+  var fields = _.extend({}, extra);
+  if (message instanceof Error) {
+    var error = message;
+    message = error.message;
+    fields.error = error;
+  }
+  this.log4js.log(level, message, fields);
```
Comment (tomaszdurka) [inline lib/logger.js]:
```js
var fields = _.extend({}, extra);
if (message instanceof Error) {
  var error = message;
  message = error.message;
  fields.error = error;
}
fields.message = message;
fields.level = level;
this.log4js.log(fields);
```

**Pull request merged**